src/main/java/io/pravega/perf/PravegaPerfTest.java

```diff
@@ -468,11 +488,43 @@ public void shutdown(long endTime) {
         }
 
         public List<ReaderWorker> getConsumers() throws URISyntaxException {
+            return batchReaders ? getBatchConsumers() : getStreamingConsumers();
+        }
+
+        public List<ReaderWorker> getBatchConsumers() {
+            final List<ReaderWorker> readers;
+            if (consumerCount > 0) {
+                List<SegmentRange> segmentRanges = streamHandle.getBatchSegmentRanges(batchClientFactory);
+
+                if (consumerCount > segmentRanges.size()) {
+                    consumerCount = segmentRanges.size();
+
+                    log.info("Limiting To {} consumers due to small number of segment ranges", consumerCount);
+                }
+
+                List<List<SegmentRange>> assignedRanges = assignSegmentsToConsumers(segmentRanges, consumerCount);
+
+                readers = IntStream.range(0, consumerCount)
+                    .boxed()
+                    .map(i ->
+                        new PravegaBatchReaderWorker(i, eventsPerConsumer,
+                        runtimeSec, startTime, consumeStats,
+                        rdGrpName, TIMEOUT, writeAndRead, batchClientFactory, assignedRanges.get(i))
+                    )
+                    .collect(Collectors.toList());
+            } else {
+                readers = null;
```

Comment:
Is this case possible? If the user does not specify `consumers`, we could take as default the number of segments or just `1`. But if the user sets `consumerCount <= 0`, the tool should throw an error even before reaching this point telling the user that the input is incorrect, right?

Reply:
This logic follows the same pattern as the streaming consumers.  The problem is that the test doesn't know if consumers are required, it only knows that because `consumers` > 0.  If `consumers` was defaulted to 1 then a test would always have a consumer, which is not desirable.